src/TSTransformer/nodes/statements/transformVariableStatement.ts

```diff
@@ -91,10 +56,10 @@ export function transformVariable(state: TransformState, identifier: ts.Identifi
 	});
 }
 
-function transformLuaTupleDestructure(
+function transformOptimizedArrayBindingPattern(
 	state: TransformState,
 	bindingPattern: ts.ArrayBindingPattern,
-	value: luau.Expression,
+	rhs: luau.Expression | luau.List<luau.Expression>,
```

Comment:
Still not great looking at the duplication between the two functions, at least inside the `for (const element of` loop recursing logic.